packages/client/hmi-client/src/services/model-templating.ts

```diff
@@ -516,14 +517,21 @@ export function updateFlattenedTemplateInView(flattenedCanvas: ModelTemplateCanv
 }
 
 // Helper function for finding port position when auto drawing junctions and edges
-function getPortPosition(templateCard: ModelTemplateCard, portId: string) {
+function getPortPosition(
+	templateCard: ModelTemplateCard,
+	portId: string,
+	potentialPortPositions?: Map<string, OffsetValues>
+) {
 	const cardWidth = 168;
+	const id = `${templateCard.id}-${portId}`;
 
 	// Default to fallback values for port position (top right of the card)
 	let x = templateCard.x + cardWidth;
 	let y = templateCard.y;
 	// Get the position of the port element
-	const portElement = document.getElementById(`${templateCard.id}-${portId}`);
+	const portElement = potentialPortPositions
+		? potentialPortPositions.get(id)
+		: document.getElementById(id);
```

Comment:
This is a bit sus - since `document.getElementById(id);` and `potentialPortPositions.get(id)` are returning very different objects that just so happens to have some overlap.

It would be clearer and easier to understand if you can wrap it with a helper function

```
getElementOffsetValues(document.getElementById(id))
```